presto-parser/src/main/java/com/facebook/presto/sql/parser/ParsingWarning.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.sql.parser;
+
+import com.facebook.presto.sql.tree.NodeLocation;
+
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class ParsingWarning
+{
+    // Avoid extending ParsingException due to cost of filling in a stacktrace on creation
+    private final int line;
+    private final int charPositionInLine;
+    private final String warning;
+
+    public ParsingWarning(String warningMessage, int line, int charPositionInLine)
+    {
+        this.warning = requireNonNull(warningMessage, "message is null");
+        this.line = line;
+        this.charPositionInLine = charPositionInLine;
+    }
+
+    public ParsingWarning(String message, NodeLocation nodeLocation)
+    {
+        this(message, nodeLocation.getLineNumber(), nodeLocation.getColumnNumber());
+    }
+
+    public ParsingWarning(String message)
+    {
+        this(message, 1, 0);
+    }
+
+    public int getLineNumber()
+    {
+        return line;
+    }
+
+    public int getColumnNumber()
+    {
+        return charPositionInLine + 1;
+    }
+
+    public String getWarning()
+    {
+        return warning;
+    }
+
+    public String getMessage()
+    {
+        return format("line %s:%s: %s", getLineNumber(), getColumnNumber(), getWarning());
```

Comment:
If you change `charPositionInLine` to `columNumber` you can use all the fields directly.